src/main/java/modify_event_use_case/ModifyEventDataAccess.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Data Access Class. Interacts with the in-memory week to "save" modifications.
+ * Author: Daniel Livshits
+ * Created: Nov 19, 2022
+ * Modified by:
+ * Last Modified: Nov 19, 2022
+ */
+package modify_event_use_case;
+
+import data_access.WeekDataAccess;
+import data_access.DataAccessEvent;
+
+import java.time.LocalTime;
+import java.util.HashMap;
+import java.util.Objects;
+
+public class ModifyEventDataAccess extends WeekDataAccess implements ModifyEventDsGateway{
+    @Override
+    public void save(ModifyEventDsInputData dataAccessInput){
+        HashMap<String, DataAccessEvent> eventsToChange = days.get(dataAccessInput.getDayIndex()).getEvents();
+        DataAccessEvent eventToChange = eventsToChange.get(dataAccessInput.getTitle());
+        DataAccessEvent newEvent = new DataAccessEvent(dataAccessInput.getNewTitle(), dataAccessInput.getNewStartTime(),
+                dataAccessInput.getNewEndTime());
+        eventsToChange.remove(eventToChange.getTitle());
```

Comment:
Assuming the start and end times of an Event object can be mutated with a set() method, I think it would be better to hold the value retuned by .remove() in a DataAccessEvent object, modify its timings, and then add that same object back to the HashMap, since you technically don't need to "delete" the event to change it.

Reply:
This may be true, but it is possible to modify the name of the event as well as the start and end times, and the current version of this method handles all possible ways to modify an event in our current implementation. I can make a separate case for when the original and new name for the event are the same where it doesn't remove the event from days, but the code segment in question would still be necessary.

Reply:
I can see why you would want to simply add a new Event instead of mutating an existing once in case all three of Event title, start time and end time are changed; this would have the same effect on the underlying data structure as Create Event.

Reply:
After reading again, I realized I misunderstood the argument made in this review. I agree with the suggested change and didn't realize it wasn't already the case before my previous response. Apologies for not fixing this earlier.